lib/ASRouterTriggerListeners.jsm

```diff
@@ -77,6 +77,64 @@ function checkURLMatch(aLocationURI, { hosts, matchPatternSet }, aRequest) {
  * have idempotent `init` and `uninit` methods.
  */
 this.ASRouterTriggerListeners = new Map([
+  [
+    "openArticleURL",
+    {
+      id: "openArticleURL",
+      _initialized: false,
+      _triggerHandler: null,
+      _hosts: new Set(),
+      _matchPatternSet: null,
+      readerModeEvent: "Reader:UpdateReaderButton",
```

Comment:
Will this also trigger when the reader button changes to be inactive? I guess that's accounted for by checking `data.isArticle`?

Reply:
Yes the event triggers on every page, even multiple times (if you have sub frames for example) and it could be `false` every time.